clients/gitlabrepo/client.go

```diff
@@ -180,11 +180,17 @@ func (client *Client) GetFileContent(filename string) ([]byte, error) {
 func (client *Client) ListCommits() ([]clients.Commit, error) {
 	// Get commits from REST API
 	commitsRaw, err := client.commits.listRawCommits()
+	var before *time.Time
+
 	if err != nil {
 		return []clients.Commit{}, err
 	}
 
-	before := commitsRaw[0].CommittedDate
+	if len(commitsRaw) < 1 {
+		return []clients.Commit{}, nil
+	} else {
+		before = commitsRaw[0].CommittedDate
+	}
```

Comment:
In the interest of keeping the [happy path left](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88), can we get rid of the `else` block?

```go
if len(commitsRaw) < 1 {
	return []clients.Commit{}, nil
} 

before := commitsRaw[0].CommittedDate
```